### PR TITLE
Added Tile Matrix Set Retrieval and Collection Tile URL Generation

### PR DESCRIPTION
**Pull request by ronitjadhav**
This pull request introduces three new functionalities to the OGC API endpoint:  

1. `getTileMatrixSets`: This method retrieves the tile matrix sets advertised by the endpoint. Tile matrix sets are a crucial part of the OGC API, as they define the grid used for map tiling. 
2. `parseTileMatrixSets`: This function is used to parse the tile matrix sets from the fetched document. It ensures the correct extraction and formatting of the tile matrix set data.  
3. `getCollectionTileUrl`: This method generates a URL for the tiles of a specified collection, with a specified tile matrix set. This allows for easy access to specific tiles within a collection. 

Tests can be performed using this ogc-api (contains vector tiles): https://maps.gnosis.earth/ogcapi/

These additions enhance the library's capabilities in handling tiled map data, providing developers with more tools to interact with OGC APIs.

### CODE REVIEW

**Comment by ronitjadhav** [comment]
I have implemented the `vectorTileCollections` method.
This method returns a Promise that resolves to an array of identifiers for collections that offer vector tiles.

The method works by first checking if the endpoint offers tiles. If it does, it fetches the collections and filters them to only include those that offer vector tiles. This is determined by checking if the `vectorTileFormats` array in the collection info is not empty. The method then returns an array of identifiers for these collections.

File: src/ogc-api/model.ts [hunk 1]
```diff
@@ -54,6 +54,10 @@ export interface OgcApiCollectionInfo {
   license?: string;
   queryables: CollectionParameter[];
   sortables: CollectionParameter[];
+  // will be empty if the endpoint does not support tiles
+  mapTileFormats: MimeType[];
+  vectorTileFormats: MimeType[];
+  supportedTileMatrixSet: string[]; // identifiers
```
Comment (jahow) [inline src/ogc-api/model.ts:60]:
```suggestion
  supportedTileMatrixSets: string[]; // identifiers
```
I think this property name would be more descriptive

File: src/ogc-api/endpoint.ts [hunk 2]
```diff
@@ -130,6 +147,29 @@ ${e.message}`);
       .then(parseCollections('feature'));
   }
 
+  /**
+   * A Promise which resolves to an array of vector tile collection identifiers as strings.
+   */
+  get vectorTileCollections(): Promise<string[]> {
```
Comment (jahow) [inline src/ogc-api/endpoint.ts:153]: I'm wondering if there would be a way to get a list of collections that support vector tiles without having to query each of them; `getCollectionInfo` will do a separate HTTP query for each and it is currently not cached, there might be a better way to get that list?

**Review by jahow — COMMENTED** [review]
Thanks! It's looking really good. I made a comment about the `vectorTilesCollections` getter, somehow I think we have to find a simpler way to get that list.

Maybe `allCollections` could return more than an identifier for each, e.g. a small object containing information like "hasVectorTiles", "hasMapTiles" etc.

**Comment by ronitjadhav** [comment]
@jahow following your suggestion, the allCollections function now returns an object.

In this update:

- The `parseCollections` function has been optimized to return an array of objects, each representing a collection. Each object contains the collection's name along with boolean flags indicating the presence of records, features, vector tiles, or map tiles, thereby facilitating enhanced functionality within `allCollections`.
- Additionally, `recordCollections`, `featureCollections`, and `vectorTileCollections` have been revised to leverage the enhancements in parseCollections.
- A new getter method, `mapTileCollections`, has been introduced to provide an array of map tile collection identifiers as strings

**Review by jahow — APPROVED** [review]
Thank you!! This is much better now 🙂 good work!

**Pull request merged**